_rules/SVG-image-has-name.md

```diff
@@ -0,0 +1,143 @@
+---
+name: SVG image has accessible name
+description: |
+  This rule checks that each SVG image element that is included in the accessibility tree, has an accessible name.
+
+success_criterion:
+- 1.1.1 # Non-Text Content
+
+test_aspects:
+- DOM Tree
+- CSS Styling
+
+authors:
+- Bryn Anderson
+---
+
+## Test procedure
+
+### Applicability
+
+The rule applies to any [SVG](https://www.w3.org/2000/svg) element with the [semantic role](#semantic-role) of `img` or `graphics-document`, that is [included in the accessibility tree](#included-in-the-accessibility-tree).
+
+### Expectation
+
+Each target element has a an accessible name.
+
+## Assumptions
+
+*There are currently no assumptions*
+
+## Accessibility support
+
+Browser and assitive technology support for SVG `title` and `desc` elements is currently inconsistent. Using WAI ARIA in combination with the `img` role for non-decorative `svg` elements significantly improves accessibility brwoser support.
+
+## Background
+
+- [Understanding Success Criterion 1.1.1: Non-text Content](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html)
+- [SVG Accessibility API Mappings](https://www.w3.org/TR/svg-aam-1.0)
+- [WAI-ARIA Graphics Module](https://www.w3.org/TR/graphics-aria-1.0/)
+- [ARIA4: Using a WAI-ARIA role to expose the role of a user interface component](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA4)
+- [ARIA6: Using aria-label to provide labels for objects](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA6)
+- [ARIA10: Using aria-labelledby to provide a text alternative for non-text content](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA10)
+
+## Test Cases
+
+### Passed
+
+#### Pass example 1
+
+The `<svg>` element has a role of `img` which takes it's accessible name from it's `<title>` element.
+
+```html
+<svg xmlns="https://www.w3.org/2000/svg" role="img" width="100" height="100">
```

Comment:
The validator rejects secure protocol `https` in the xmlns attribute. It says it only accept `http` (honestly I don't why, what, how ^^ I just find that error). 

````html
<svg xmlns="http://www.w3.org/2000/svg" role="img" width="100" height="100">

Reply:
I have removed the "s" assuming that there is no consequence to doing so.